### PR TITLE
Converted tests to XCTest

### PR DESCRIPTION
**Pull request by bamse16**
Converted the unit tests from SenTestingKit to XCTest as SenTestingKit was deprecated a couple of SDK versions ago.

### CODE REVIEW

File: Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme [hunk 1]
```diff
@@ -23,10 +23,10 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme]: 2 changes just shuffled

File: Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme [hunk 2]
```diff
@@ -39,24 +39,36 @@
             </BuildableReference>
          </TestableReference>
       </Testables>
+      <AdditionalOptions>
+      </AdditionalOptions>
    </TestAction>
    <LaunchAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme]: shuffled

File: Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme [hunk 3]
```diff
@@ -39,24 +39,36 @@
...
+            BuildableName = "QueryKit.framework"
+            BlueprintName = "QueryKit"
+            ReferencedContainer = "container:QueryKit.xcodeproj">
+         </BuildableReference>
+      </MacroExpansion>
       <AdditionalOptions>
       </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
+      buildConfiguration = "Release"
```
Comment (dmoagx) [inline Frameworks/QueryKit/QueryKit.xcodeproj/xcshareddata/xcschemes/QueryKit.xcscheme]: shuffled

File: UnitTests/SPDataAdditionsTests.m [hunk 4]
```diff
@@ -291,85 +291,102 @@ - (void)testEnumerateLinesBreakingAt_withBlock
...
+		NSValue *line2 = [NSValue valueWithRange:NSMakeRange(4, 5)];
+
 		__block NSUInteger invocations = 0;
 		NSData *data = [NSData dataWithBytes:inp length:strlen(inp)];
 		[data enumerateLinesBreakingAt:SPLineTerminatorAny withBlock:^(NSRange line, BOOL *stop) {
+			NSValue *lineValue = [NSValue valueWithRange:line];
 			switch (invocations) {
 				case 0:
-					STAssertEquals(line, NSMakeRange(0, 3), @"range of first line");
+					XCTAssertTrue([lineValue isEqualToValue:line1], @"range of first line");
```
Comment (dmoagx) [inline UnitTests/SPDataAdditionsTests.m]: maybe use NSEqualRanges instead?

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro (10.6 SDK).xcscheme [hunk 5]
```diff
@@ -48,17 +48,21 @@
             ReferencedContainer = "container:sequel-pro.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
+      <AdditionalOptions>
+      </AdditionalOptions>
    </TestAction>
    <LaunchAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro (10.6 SDK).xcscheme]: shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro (10.6 SDK).xcscheme [hunk 6]
```diff
@@ -71,12 +75,13 @@
       </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
+      buildConfiguration = "Release"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro (10.6 SDK).xcscheme]: shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme [hunk 7]
```diff
@@ -32,17 +32,17 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme]: 2x shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme [hunk 8]
```diff
@@ -57,15 +57,18 @@
             ReferencedContainer = "container:sequel-pro.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
+      <AdditionalOptions>
+      </AdditionalOptions>
    </TestAction>
    <LaunchAction
+      buildConfiguration = "Distribution"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme]: shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme [hunk 9]
```diff
@@ -81,10 +84,10 @@
       </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
+      buildConfiguration = "Release"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro Release Build (10.6+).xcscheme]: shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme [hunk 10]
```diff
@@ -23,17 +23,17 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme]: 2x shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme [hunk 11]
```diff
@@ -55,17 +55,21 @@
             isEnabled = "YES">
          </EnvironmentVariable>
       </EnvironmentVariables>
+      <AdditionalOptions>
+      </AdditionalOptions>
    </TestAction>
    <LaunchAction
+      buildConfiguration = "Debug"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme]: shuffle

File: sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme [hunk 12]
```diff
@@ -78,12 +82,13 @@
       </AdditionalOptions>
    </LaunchAction>
    <ProfileAction
+      buildConfiguration = "Release"
```
Comment (dmoagx) [inline sequel-pro.xcodeproj/xcshareddata/xcschemes/Sequel Pro.xcscheme]: shuffle

**Comment by bamse16** [comment]
Updated code after Max's suggestions.

**Pull request merged**